real_time/surface_src/java_src/Alice/src/main/java/com/roboclub/robobuggy/ui/LoggingPanel.java

```diff
@@ -0,0 +1,122 @@
+package com.roboclub.robobuggy.ui;
+
+import java.awt.Color;
+import java.awt.Font;
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+import java.sql.NClob;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+import javax.swing.JButton;
+import javax.swing.JFormattedTextField;
+import javax.swing.JLabel;
+import javax.swing.SwingConstants;
+import javax.swing.Timer;
+
+import com.roboclub.robobuggy.logging.RobotLogger;
+import com.roboclub.robobuggy.main.RobobuggyMessageLevel;
+import com.roboclub.robobuggy.main.RobobuggyConfigFile;
+import com.roboclub.robobuggy.main.RobobuggyLogicException;
+import com.roboclub.robobuggy.messages.GuiLoggingButtonMessage;
+import com.roboclub.robobuggy.ros.NodeChannel;
+import com.roboclub.robobuggy.ros.Publisher;
+
+public class LoggingPanel extends RoboBuggyGUIContainer{
+	private static JButton play_btn;
+	private JFormattedTextField time_lbl;
+	private static final int TIME_ZONE_OFFSET = 18000000;//5 hours
+	private static Date startTime;
+	private static Timer timer;
+	Publisher logging_button_pub;
```

Comment:
![Codacy](https://www.codacy.com/versioned/images/favicon.png) Issue found: [Use explicit scoping instead of the default package private level](https://www.codacy.com/app/Mr-Trevor-Decker/RoboBuggy/file/1814872849/issues/source?bid=2938665&fileBranchId=3059924#l31)